chain/near.ts

```diff
@@ -0,0 +1,416 @@
+import { Bytes } from '../common/collections'
+import { BigInt } from '../common/numbers'
+
+// Most types from this namespace are direct mappings or adaptations from:
+// https://github.com/streamingfast/proto-near/blob/develop/sf/near/codec/v1/codec.proto
+export namespace near {
+  export type CryptoHash = Bytes
+
+  export type AccountId = string
+
+  export type BlockHeight = u64
+
+  export type Balance = BigInt
+
+  export type Gas = u64
+
+  export type ShardId = u64
+
+  export type NumBlocks = u64
+
+  export type ProtocolVersion = u32
+
+  export type Payload = u64
+
+  export enum CurveKind {
+    ED25519 = 0,
+    SECP256K1 = 1,
+  }
+
+  export class Signature {
+    constructor(
+      public kind: CurveKind,
+      public bytes: Bytes,
+    ) {}
+  }
+
+  export class PublicKey {
+    constructor(
+      public kind: CurveKind,
+      public bytes: Bytes,
+    ) {}
+  }
+
+  export enum AccessKeyPermissionKind {
+    FUNCTION_CALL = 0,
+    FULL_ACCESS = 1,
+  }
+
+  export class FunctionCallPermission {
+    constructor(
+      public allowance: BigInt,
+      public receiverId: string,
+      public methodNames: Array<string>,
+    ) {}
+  }
+
+  export class FullAccessPermission {}
+
+  export class AccessKeyPermissionValue {
+    constructor(
+      public kind: AccessKeyPermissionKind,
+      public data: Payload,
+    ) {}
+
+    toFunctionCall(): FunctionCallPermission {
+      assert(this.kind == AccessKeyPermissionKind.FUNCTION_CALL, 'AccessKeyPermissionValue is not a \'FunctionCall\'.')
+      return changetype<FunctionCallPermission>(this.data as u32)
+    }
+
+    toFullAccess(): FullAccessPermission {
+      assert(this.kind == AccessKeyPermissionKind.FULL_ACCESS, 'AccessKeyPermissionValue is not a \'FullAccess\'.')
+      return changetype<FullAccessPermission>(this.data as u32)
+    }
+
+    static fromFunctionCall(input: FunctionCallPermission): AccessKeyPermissionValue {
+      return new AccessKeyPermissionValue(AccessKeyPermissionKind.FUNCTION_CALL, changetype<u32>(input))
+    }
+
+    static fromFullAccess(input: FullAccessPermission): AccessKeyPermissionValue {
+      return new AccessKeyPermissionValue(AccessKeyPermissionKind.FULL_ACCESS, changetype<u32>(input))
+    }
+  }
+
+  export class AccessKey {
+    constructor(
+      public nonce: u64,
+      public permission: AccessKeyPermissionValue,
+    ) {}
+  }
+
+  export class DataReceiver {
+    constructor(
+      public dataId: CryptoHash,
+      public receiverId: string,
+    ) {}
+  }
+
+  export enum ActionKind {
+    CREATE_ACCOUNT = 0,
+    DEPLOY_CONTRACT = 1,
+    FUNCTION_CALL = 2,
+    TRANSFER = 3,
+    STAKE = 4,
+    ADD_KEY = 5,
+    DELETE_KEY = 6,
+    DELETE_ACCOUNT = 7,
+  }
+
+  export class CreateAccountAction {}
+
+  export class DeployContractAction {
+    constructor(
+      public code: string,
+    ) {}
+  }
+
+  export class FunctionCallAction {
+    constructor(
+      public methodName: string,
+      public args: string,
```

Comment:
This one also I think we should turn it into a `Bytes` directly (again that I would base64 decode in graph-node until native in the protobuf definition).

Let's discuss this one at our next meeting.